zio-http/src/main/scala/zio/http/Route.scala

```diff
@@ -0,0 +1,486 @@
+package zio.http
+
+import io.netty.handler.codec.http.HttpHeaderNames
+import zio._
+import zio.http.model.{Headers, MediaType, Status}
+import zio.http.socket.{SocketApp, WebSocketChannelEvent}
+import zio.stream.ZStream
+
+import java.io.{File, FileNotFoundException}
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+trait Route[-R, +Err, -In, +Out] { self =>
```

Comment:
As strange as it might be, executable encodings like this one can sometimes be slower than the declarative encoding due to dynamic dispatch. Possibly worth making this `sealed` in case we want to change our minds later.

Reply:
I was thinking about it, but then decided its faster to build up the new model with executable encoding (as I was not sure how exactly it will look like). 

So let's make it `sealed` now, and experiment with declarative encoding in a follow-up PR if we decide to merge this?

Reply:
Yes!